### PR TITLE
[23.1] Encode tool link consistently

### LINKED ISSUE #17425: Tool link is not URL encoded
**Describe the bug**

If one copies a link then spaces in tool IDs are not encoded. Guess they should. 

Example `https://galaxy.intranet.ufz.de/root?tool_id=toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: transeq101/5.0.0`

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 23.1

### PR DESCRIPTION
**Pull request by davelopez**
Fixes #17425

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/Tool/utilities.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ import { getAppRoot } from "onload/loadConfig";
 import { copy } from "utils/clipboard";
 
 export function copyLink(toolId, message) {
-    copy(`${window.location.origin + getAppRoot()}root?tool_id=${toolId}`, message);
+    const link = `${window.location.origin + getAppRoot()}root?tool_id=${toolId}`;
+    // Encode the link to handle special characters in tool id
+    copy(encodeURI(link), message);
```
Comment (mvdbeek) [inline client/src/components/Tool/utilities.js:7]: Are you sure that doesn't break tools with `+` in the tool id ? Can we maybe just replace space with percent encoding ?

Comment (davelopez) [reply client/src/components/Tool/utilities.js:7]: Good point. I just checked and `encodeURI` should be fine with `+` according to the docs in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs._encodeuricomponent

I will add another test just to make 100% sure.

Any other possible special character that can be contained in a tool ID that we should check?

**Review by mvdbeek — APPROVED** [review]
Bonus points if you can go through all tool ids in https://usegalaxy.org/api/tools?in_panel=false and check that they all resolve correctly ? I don't know if that's necessary but it's been a test we've used in similar cases.

**Comment by davelopez** [comment]
You know I love some good old Bonus points :laughing: 
But you mean in a script somewhere else not as part of this PR right? :thinking:

**Comment by mvdbeek** [comment]
haha, yes!

**Comment by mvdbeek** [comment]
super useful, thanks a lot!

**Comment by mvdbeek** [comment]
How did your test turn out ?

**Comment by davelopez** [comment]
Ouch, I forgot to report back :sweat_smile: 

Here is the result, 4976 tool IDs were checked against .org, and all of them are OK :tada: 

[tool_ids_encoding_check_main.csv](https://github.com/galaxyproject/galaxy/files/14206419/tool_ids_encoding_check_main.csv)

**Pull request merged**